### PR TITLE
PLT-8379/#67: Add a Dev Talks Section to Developers Site

### PR DESCRIPTION
**Pull request by MusikPolice**
Added a new page to the contribute section that lists all videos from our Dev Talks playlist on YouTube. This page is generated from a data file that lists each video's id, title, and description. A script has been added to the Makefile that uses the YouTube API to auto-generate the contents of the data file.

**Ticket Link:**
https://github.com/mattermost/mattermost-developer-documentation/issues/67
https://mattermost.atlassian.net/browse/PLT-8379

**Note:** For this to work on Travis CI, the `YOUTUBE_API_KEY` environment variable needs to be set. I have created an API key under my Mattermost account that can be used.

### CODE REVIEW

**Comment by ccbrown** [comment]
Looks fine to me and I'm not sure about your comment, but I do have one thought:

Rather than committing a JSON file that's going to inevitably get out-dated, can we show a placeholder with instructions for how to generate it like we do for our other data?

File: site/layouts/shortcodes/devtalks.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div class="youtube-videos">
+    {{ range $.Site.Data.devtalks.youtube.videos }}
```
[the remark this answers is not in the document]
Comment (jwilander) [reply site/layouts/shortcodes/devtalks.html]: Not really sure how to do that either ¯\\\_(ツ)_/¯

**Comment by MusikPolice** [comment]
@ccbrown that's a good idea. Done.

**Comment by ccbrown** [comment]
Should probably also add site/data/devtalks/youtube.json to the .gitignore now too.

**Comment by MusikPolice** [comment]
@ccbrown the idea was that there might be more than one data file powering dev talks, but in retrospect, that isn't true. Changed it to be more in line with other files, added it to gitignore as requested.

**Pull request merged**